packages/core/contracts/common/implementation/FixedPoint.sol

```diff
@@ -373,4 +377,386 @@ library FixedPoint {
+     */
+    function mulCeil(Signed memory a, int256 b) internal pure returns (Signed memory) {
+        // Since b is an int, there is no risk of truncation and we can just mul it normally
+        return Signed(a.rawValue.mul(b));
+    }
+
+    /**
+     * @notice Divides one `Signed` by an `Signed`, reverting on overflow or division by 0.
+     * @dev This will "floor" the quotient.
+     * @param a a FixedPoint numerator.
+     * @param b a FixedPoint denominator.
+     * @return the quotient of `a` divided by `b`.
+     */
+    function div(Signed memory a, Signed memory b) internal pure returns (Signed memory) {
+        // There are two caveats with this computation:
+        // 1. Max value for the number dividend `a` represents is ~10^41, otherwise an intermediate value overflows.
+        // 10^41 is stored internally as an int256 10^59.
+        // 2. Results that can't be represented exactly are truncated not rounded. E.g., 2 / 3 = 0.6 repeating, which
+        // would round to 0.666666666666666667, but this computation produces the result 0.666666666666666666.
+        return Signed(a.rawValue.mul(SFP_SCALING_FACTOR).div(b.rawValue));
+    }
+
+    /**
+     * @notice Divides one `Signed` by an unscaled int256, reverting on overflow or division by 0.
+     * @dev This will "floor" the quotient.
+     * @param a a FixedPoint numerator.
+     * @param b an int256 denominator.
+     * @return the quotient of `a` divided by `b`.
+     */
+    function div(Signed memory a, int256 b) internal pure returns (Signed memory) {
+        return Signed(a.rawValue.div(b));
+    }
+
+    /**
+     * @notice Divides one unscaled int256 by an `Signed`, reverting on overflow or division by 0.
+     * @dev This will "floor" the quotient.
+     * @param a an int256 numerator.
+     * @param b a FixedPoint denominator.
+     * @return the quotient of `a` divided by `b`.
+     */
+    function div(int256 a, Signed memory b) internal pure returns (Signed memory) {
+        return div(fromUnscaledInt(a), b);
+    }
+
+    /**
+     * @notice Divides one `Signed` by an `Signed` and "ceil's" the quotient, reverting on overflow or division by 0.
+     * @param a a FixedPoint numerator.
+     * @param b a FixedPoint denominator.
+     * @return the quotient of `a` divided by `b`.
+     */
+    function divAwayFromZero(Signed memory a, Signed memory b) internal pure returns (Signed memory) {
+        int256 aScaled = a.rawValue.mul(SFP_SCALING_FACTOR);
+        int256 divTowardsZero = aScaled.div(b.rawValue);
+        // Manual mod because SignedSafeMath doesn't support it.
+        int256 mod = aScaled % b.rawValue;
+        if (mod != 0) {
+            bool isResultPositive = isLessThan(a, 0) == isLessThan(b, 0);
+            int256 valueToAdd = isResultPositive ? int256(1) : int256(-1);
+            return Signed(divTowardsZero.add(valueToAdd));
+        } else {
+            return Signed(divTowardsZero);
+        }
+    }
```

Comment:
@nicholaspai notice the changes in this function. It needed to become a bit more complex to do "away from zero" rather than just ceil. The reason it was done this way is that solidity's division floors _towards zero_, not down, so this function was made to be the inverse of that.

Reply:
So to summarize my understanding:
- if a / b is floored towards 0:
- - if the result should be positive, add 1 "away from zero"
- - if the result is negative, subtract 1 "away from  zero"

Overall:
- divCeil is replaced by divAwayFromZero (I think this is more intuitive naming +1)
- same with mul

Reply:
Exactly. The reason we don't need to change the names in the unsigned case is 1) it would require messing with existing smart contracts and 2) away from zero and up are the same direction in all cases.